lib/pbench/test/unit/server/conftest.py

```diff
@@ -736,31 +700,131 @@ def create_drb_user(client, server_config, fake_email_validator):
 
 
 @pytest.fixture()
-def pbench_token(client, server_config, create_drb_user):
-    # Login user to get valid pbench token
-    response = login_user(client, server_config, "drb", generic_password)
-    assert response.status_code == HTTPStatus.OK
-    data = response.json
-    assert data["auth_token"]
-    return data["auth_token"]
+def pbench_admin_token(client, create_admin_user):
+    return generate_token(
+        user=create_admin_user,
+        username=admin_username,
+        pbench_client_roles=["ADMIN"],
+    )
+
+
+@pytest.fixture()
+def pbench_token(client, create_drb_user):
+    """
+    OIDC valid token for the 'drb' user.
+    """
+    return generate_token(username="drb", user=create_drb_user)
+
+
+@pytest.fixture()
+def pbench_token_invalid(client, create_drb_user):
+    """
+    OIDC invalid token for the 'drb' user
+    """
+    return generate_token(username="drb", user=create_drb_user, valid=False)
+
+
+def generate_token(
+    username: str,
+    user: Optional[User] = None,
+    pbench_client_roles: Optional[list[str]] = None,
+    valid: Optional[bool] = True,
```

Comment:
I think Webb's explanation is a bit murkier than necessary, but, yeah, the type hint allows a call with `valid=None`, which would in fact be pointless because the only use of `valid` in the code is `if valid`, meaning that `valid=False` and `valid=None` are functionally identical and the latter would be a bit silly. I suspect in this case we should be using `valid: bool = True`.

Reply:
> I think Webb's explanation is a bit murkier than necessary

Bummer!

> I suspect in this case we should be using `valid: bool = True`.

Yes...that's what I was trying to say.